source/server/http_test_server_filter.h

```diff
@@ -2,20 +2,68 @@
 
 #include <string>
 
+#include "envoy/common/time.h"
 #include "envoy/server/filter_config.h"
 
+#include "external/envoy/source/common/common/lock_guard.h"
+#include "external/envoy/source/common/common/thread.h"
+
 #include "api/server/response_options.pb.h"
 
 namespace Nighthawk {
 namespace Server {
 
-// Basically this is left in as a placeholder for further configuration.
+/**
+ * Filter configuration container class for the test server extension.
+ * Instances of this class will be shared accross instances of HttpTestServerDecoderFilter.
+ */
 class HttpTestServerDecoderFilterConfig {
 public:
+  /**
+   * Constructs a new HttpTestServerDecoderFilterConfig instance.
+   *
+   * @param proto_config The proto configuration of the filter.
+   */
   HttpTestServerDecoderFilterConfig(nighthawk::server::ResponseOptions proto_config);
+
+  /**
+   * @return const nighthawk::server::ResponseOptions& read-only reference to the proto config
+   * object.
+   */
   const nighthawk::server::ResponseOptions& server_config() { return server_config_; }
 
+  /**
+   * Gets the number of elapsed nanoseconds since the last call (server wide).
+   * Safe to use concurrently.
+   *
+   * @param time_source Time source that will be used to obain an updated monotonic time sample.
+   * @return uint64_t 0 on the first call, else the number of elapsed nanoseconds since the last
+   * call.
+   */
+  static uint64_t getElapsedNanosSinceLastRequest(Envoy::TimeSource& time_source);
+
 private:
+  /**
+   *  Utility class for thread safe tracking of elapsed monotonic time.
+   */
+  class ThreadSafeMontonicTimeStopwatch {
+  public:
+    ThreadSafeMontonicTimeStopwatch() : start_(Envoy::MonotonicTime::min()) {}
+    /**
+     * @param time_source used to obtain a sample of the current monotonic time.
+     * @return uint64_t 0 on the first invocation, and the number of elapsed nanoseconds since the
+     * last invocation otherwise.
+     */
+    uint64_t getElapsedNsAndReset(Envoy::TimeSource& time_source);
+
+  private:
+    Envoy::Thread::MutexBasicLockable lock_;
+    Envoy::MonotonicTime start_ GUARDED_BY(lock_);
+  };
+
+  static ThreadSafeMontonicTimeStopwatch& lastRequestStopwatch() {
```

Comment:
Can we document this?

Note the current name suggests that maybe there are multiple of these, new ones created for each new "last request". Unless I misunderstood something, I think this returns a global singleton. If so, can we mention the fact in documentation and optionally try to find another name, e.g. `getRequestStopwatch()` or similar.